oidc-exchange.py

```diff
@@ -0,0 +1,157 @@
+import os
+import sys
+from pathlib import Path
+from typing import NoReturn
+from urllib.parse import urlparse
+
+import id  # pylint: disable=W0622
```

Comment:
I have a rule that pylint ignores should use named identifiers. This let's us skip writing an explanatory comment to save the reader from having to google what this is.
```suggestion
import id  # pylint: disable=redefined-builtin
```